### PR TITLE
KAFKA-14484: [2/N] Move RequestLocal to storage module

### PR DESCRIPTION
**Pull request by alok123t**
This PR simply moves the `RequestLocal` to storage module

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. This shouldn't be in in the storage module - it doesn't have anything to do with storage.

**Comment by alok123t** [comment]
> Thanks for the PR. This shouldn't be in in the storage module - it doesn't have anything to do with storage.

@ijuma this is a dependency for moving the UnifiedLog to storage, shall we move to `sever-common` or any suggestions?

**Comment by ijuma** [comment]
I think you can change the parameter to be BufferSupplier instead of RequestLocal.

**Comment by alok123t** [comment]
> I think you can change the parameter to be BufferSupplier instead of RequestLocal.

@ijuma thanks for the pointer, updated

File: core/src/main/scala/kafka/cluster/Partition.scala [hunk 1]
```diff
@@ -1302,7 +1302,7 @@ class Partition(val topicPartition: TopicPartition,
   }
 
   def appendRecordsToLeader(records: MemoryRecords, origin: AppendOrigin, requiredAcks: Int,
-                            requestLocal: RequestLocal, verificationGuard: VerificationGuard = VerificationGuard.SENTINEL): LogAppendInfo = {
+                            bufferSupplier: BufferSupplier, verificationGuard: VerificationGuard = VerificationGuard.SENTINEL): LogAppendInfo = {
```
Comment (ijuma) [inline core/src/main/scala/kafka/cluster/Partition.scala:1305]: Hmm, why are we changing classes that we don't intend to move to the storage module?

**Comment by mimaison** [comment]
We moved `RequestLocal` to the server-common module in https://github.com/apache/kafka/commit/5fd7ce2ace1e0418df033dd32a470d4306a211ea, closing

**Pull request closed without merging**